tieredhashing/tiered_hashing.go

```diff
@@ -0,0 +1,384 @@
+package tieredhashing
+
+import (
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/asecurityteam/rolling"
+
+	"github.com/patrickmn/go-cache"
+
+	"github.com/serialx/hashring"
+)
+
+// TODO Make env vars for tuning
```

Comment:
file an issue for this if it's not in the PR

Reply:
https://github.com/filecoin-saturn/caboose/pull/86

Reply:
Done.